### PR TITLE
T-48 Rename tests and format code according guidelines

### PR DESCRIPTION
**Pull request by goremukin**

### CODE REVIEW

File: PortfolioKeeper.Tests/BusinessLogic/Parsers/BgpbInfoParserTests.cs [hunk 1]
```diff
@@ -15,20 +15,18 @@ public class BgpbInfoParserTests
...
             factory = new Mock<IExchangeRateFactory>();
-            serializer = new Moq.Mock<IBgpbSerializer>();
-
+            serializer = new Mock<IBgpbSerializer>();
             underTest = new BgpbInfoParser(factory.Object, serializer.Object);
         }
 
         [Test]
-        public void BgpbInfoParser_When_factory_Is_Null_should_Throw_ArgumentNullExcpetion()
+        public void BgpbInfoParser_When_factory_Is_Null_Should_Throw_ArgumentNullExcpetion()
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Parsers/BgpbInfoParserTests.cs]: factory->Factory

File: PortfolioKeeper.Tests/BusinessLogic/Parsers/BsbInfoParserTests.cs [hunk 2]
```diff
@@ -13,19 +13,17 @@ namespace PortfolioKeeper.Tests.BusinessLogic.Parsers
...
         public void SetUp()
         {
             factory = new Mock<IExchangeRateFactory>();
-
             underTest = new BsbInfoParser(factory.Object);
         }
 
         [Test]
-        public void BsbInfoParser_When_factory_Is_Null_should_Throw_ArgumentNullExcpetion()
+        public void BsbInfoParser_When_factory_Is_Null_Should_Throw_ArgumentNullExcpetion()
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Parsers/BsbInfoParserTests.cs]: factory->Factory

File: PortfolioKeeper.Tests/BusinessLogic/Services/AssetServiceTests.cs [hunk 3]
```diff
@@ -130,6 +126,7 @@ public void GetAssetProperties_When_Asset_Is_Not_Found_Should_Throw_ArgumentExce
         {
             // arrange
             int testId = 10;
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Services/AssetServiceTests.cs]: move it up - create constant in test class and use it

File: PortfolioKeeper.Tests/BusinessLogic/Services/BgpbInfoServiceTests.cs [hunk 4]
```diff
@@ -27,7 +26,6 @@ public void SetUp()
             bgpbProvider = new Mock<IBankInfoProvider>();
             bgpbParser = new Mock<IBankInfoParser>();
             bankInfoPersister = new Mock<IBankInfoPersister>();
-
             underTest = new BgpbInfoService(bgpbProvider.Object, bgpbParser.Object, bankInfoPersister.Object);
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Services/BgpbInfoServiceTests.cs]: bgpbParser.Object, bankInfoPersister.Object - to new lines

File: PortfolioKeeper.Tests/BusinessLogic/Services/BgpbInfoServiceTests.cs [hunk 5]
```diff
@@ -150,6 +148,7 @@ public void GetAndStoreExchangeRates_When_Called_Should_Call_Provider_Parser_And
                                 ""dt"": -0.002
                               }]";
             #endregion
+
             var rates = new List<ExchangeRateDto>
             {
                 new ExchangeRateDto { BuyRate = 1.995M, FromCurrency = "BYN", RateDay = DateTime.UtcNow, SellRate = 2.007M, ToCurrency = "USD"},
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Services/BgpbInfoServiceTests.cs]: { BuyRate = 1.995M, FromCurrency = "BYN", ... to new lines

File: PortfolioKeeper.Tests/BusinessLogic/Services/BsbInfoServiceTests.cs [hunk 6]
```diff
@@ -126,6 +124,7 @@ public void GetAndStoreExchangeRates_When_Called_Should_Call_Provider_Parser_And
                     </from_time>
                 </trader>";
             #endregion
+
             var rates = new List<ExchangeRateDto>
             {
                 new ExchangeRateDto { BuyRate = 1.992M, FromCurrency = "BYN", RateDay = DateTime.UtcNow, SellRate = 2.004M, ToCurrency = "USD"},
```
Comment (stas-litvinenka) [inline PortfolioKeeper.Tests/BusinessLogic/Services/BsbInfoServiceTests.cs]: to new lines

File: PortfolioKeeper/BusinessLogic/Helpers/ExchangeRateProvider.cs [hunk 7]
```diff
@@ -31,7 +35,7 @@ public decimal GetBestExchangeRate(char currencyId, bool buyRate = true)
             var best = exchangeRateRepository
                 .Get(e => e.RateDate.Date == lastRate.RateDate.Date,
                     orderBy: q => q.OrderByDescending(e => buyRate ? e.BuyRate : e.SellRate))
-                .FirstOrDefault();
+                .First();
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Helpers/ExchangeRateProvider.cs]: if there will be no records this code will be broken

File: PortfolioKeeper/BusinessLogic/Parsers/Bgpb/RateTypeConverter.cs [hunk 8]
```diff
@@ -16,12 +16,15 @@ public override object ReadJson(JsonReader reader, Type t, object existingValue,
             }
 
             var value = serializer.Deserialize<string>(reader);
+
             switch (value)
             {
                 case "PCC":
                     return RateTypeValue.BuyRate;
+
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Parsers/Bgpb/RateTypeConverter.cs]: no need in extra lines between cases

File: PortfolioKeeper/BusinessLogic/Parsers/Bgpb/RateTypeConverter.cs [hunk 9]
```diff
@@ -32,18 +35,22 @@ public override void WriteJson(JsonWriter writer, object untypedValue, JsonSeria
...
             }
 
             var value = (RateTypeValue)untypedValue;
+
             switch (value)
             {
                 case RateTypeValue.SellRate:
                     serializer.Serialize(writer, "SCC");
                     break;
+
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Parsers/Bgpb/RateTypeConverter.cs]: no need in extra lines

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 10]
```diff
@@ -30,16 +29,26 @@ public class AssetService : IAssetService
...
-            this.assetMapper = assetMapper ?? throw new ArgumentNullException(nameof(assetMapper));
+            this.unitOfWork = unitOfWork
+                              ?? throw new ArgumentNullException(nameof(unitOfWork));
+            this.mapper = mapper
+                          ?? throw new ArgumentNullException(nameof(mapper));
+            this.dboFactory = dboFactory
+                              ?? throw new ArgumentNullException(nameof(dboFactory));
+            this.assetMapper = assetMapper
+                               ?? throw new ArgumentNullException(nameof(assetMapper));
+
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: remove empty line

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 11]
```diff
@@ -131,45 +144,65 @@ public IEnumerable<AssetTypeDto> GetParentAssetTypes()
...
             allProperties.AddRange(parentProperties);
+
             return assetMapper.MapPropertyType(allProperties);
         }
 
         public void Create(List<AssetPropertyValueViewModel> properties, string userId, string assetName, byte assetTypeId)
         {
             if (properties == null)
+            {
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: remove brackets here.  if there is only exception throw - no need in {}

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 12]
```diff
@@ -32,36 +35,47 @@ public AccountController(AppUserManager userManager, AppSignInManager signInMana
...
+
             return View();
         }
 
         [HttpPost]
         [AllowAnonymous]
         [ValidateAntiForgeryToken]
-        public async Task<ActionResult> Login(LoginViewModel model, string returnUrl)
+        public async Task<ActionResult> Login(LoginViewModel model,
+            string returnUrl)
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: no need in extr line here

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 13]
```diff
@@ -32,36 +35,47 @@ public AccountController(AppUserManager userManager, AppSignInManager signInMana
...
             }
+
             if (!userManager.IsEmailConfirmed(user.Id))
             {
                 return View("EmailNotConfirmed", null, new EmailNotConfirmedViewModel { Email = user.Email });
             }
-            var result = await signInManager.PasswordSignInAsync(model.UserName, model.Password, model.RememberMe, shouldLockout: true);
+
+            var result = await signInManager.PasswordSignInAsync(model.UserName,
+                model.Password,
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: no need to put each arg to new line

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 14]
```diff
@@ -100,16 +116,23 @@ public ActionResult Register()
...
                 return View(model);
             }
 
             string code = await userManager.GenerateEmailConfirmationTokenAsync(user.Id);
-            var callbackUrl = Url.Action("ConfirmEmail", "Account", new { userId = user.Id, code = code }, protocol: Request.Url.Scheme);
-            await userManager.SendEmailAsync(user.Id, "Confirm your account", "Please confirm your account by clicking <a href=\"" + callbackUrl + "\">here</a>");
+            var callbackUrl = Url.Action("ConfirmEmail", "Account", new { userId = user.Id, code }, protocol: Request?.Url?.Scheme);
+
+            await userManager.SendEmailAsync(user.Id,
+                "Confirm your account",
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: no need to put it to new line

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 15]
```diff
@@ -118,34 +141,17 @@ public ActionResult Register()
...
-            };
-            portfolioService.Create(portfolioDto);
-
-            return result;
-        }
-
         [AllowAnonymous]
-        public async Task<ActionResult> ConfirmEmail(string userId, string code)
+        public async Task<ActionResult> ConfirmEmail(string userId,
+            string code)
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: no need in extra line

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 16]
```diff
@@ -170,14 +176,21 @@ public ActionResult ForgotPassword()
...
                 return View("ForgotPasswordConfirmation");
             }
+
             string code = await userManager.GeneratePasswordResetTokenAsync(user.Id);
-            var callbackUrl = Url.Action("ResetPassword", "Account", new { userId = user.Id, code = code }, protocol: Request.Url.Scheme);
-            await userManager.SendEmailAsync(user.Id, "Reset Password", "Please reset your password by clicking <a href=\"" + callbackUrl + "\">here</a>");
+            var callbackUrl = Url.Action("ResetPassword", "Account", new { userId = user.Id, code }, Request?.Url?.Scheme);
+
+            await userManager.SendEmailAsync(user.Id,
+                "Reset Password",
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: no need in extra line

File: PortfolioKeeper/BusinessLogic/Helpers/ExchangeRateProvider.cs (hunk 7 above)
Comment (goremukin) [reply PortfolioKeeper/BusinessLogic/Helpers/ExchangeRateProvider.cs]: @stas-litvinenka, Above is a check that ensures that the items are existing

**Comment by goremukin** [comment]
[T-48 Rename tests and format code according guidelines](https://trello.com/c/K5sE49TF/49-t-48-rename-tests-and-format-code-according-guidelines)

File: PortfolioKeeper/Controllers/AccountController.cs (hunk 13 above)
Comment (goremukin) [reply PortfolioKeeper/Controllers/AccountController.cs]: @stas-litvinenka, Very long line

File: PortfolioKeeper.Tests/BusinessLogic/Services/BgpbInfoServiceTests.cs [hunk 17]
```diff
@@ -150,11 +150,32 @@ public void GetAndStoreExchangeRates_When_Called_Should_Call_Provider_Parser_And
...
+                new ExchangeRateDto
+                {
+                    BuyRate = 1.995M,
+                    FromCurrency = "BYN",
+                    RateDay = DateTime.UtcNow,
+                    SellRate = 2.007M,
+                    ToCurrency = "USD"
+                },
+                new ExchangeRateDto
+                    { BuyRate = 2.305M,
```
Comment (EdwOK) [inline PortfolioKeeper.Tests/BusinessLogic/Services/BgpbInfoServiceTests.cs]: Tabulation

File: PortfolioKeeper.Tests/BusinessLogic/Services/BsbInfoServiceTests.cs [hunk 18]
```diff
@@ -126,11 +124,32 @@ public void GetAndStoreExchangeRates_When_Called_Should_Call_Provider_Parser_And
...
                 </trader>";
             #endregion
+
             var rates = new List<ExchangeRateDto>
             {
-                new ExchangeRateDto { BuyRate = 1.992M, FromCurrency = "BYN", RateDay = DateTime.UtcNow, SellRate = 2.004M, ToCurrency = "USD"},
-                new ExchangeRateDto { BuyRate = 2.304M, FromCurrency = "BYN", RateDay = DateTime.UtcNow, SellRate = 2.324M, ToCurrency = "EUR"},
-                new ExchangeRateDto { BuyRate = .03146M, FromCurrency = "BYN", RateDay = DateTime.UtcNow, SellRate = .03187M, ToCurrency = "RUB"}
+                new ExchangeRateDto
+                    { BuyRate = 1.992M,
```
Comment (EdwOK) [inline PortfolioKeeper.Tests/BusinessLogic/Services/BsbInfoServiceTests.cs]: Tabulation

File: PortfolioKeeper/BusinessLogic/Mappers/AssetServiceMapper.cs [hunk 19]
```diff
@@ -11,9 +11,11 @@ public class AssetServiceMapper: IAssetServiceMapper
...
         {
             var assetTypeDtos = new List<AssetTypeDto>();
+
             foreach (var assetType in assetTypes)
             {
                 var assetPropertyTypeDtos = new List<AssetPropertyTypeDto>();
+
                 foreach (var propertyType in assetType.PropertyTypes)
                 {
                     assetPropertyTypeDtos.Add(new AssetPropertyTypeDto
```
Comment (EdwOK) [inline PortfolioKeeper/BusinessLogic/Mappers/AssetServiceMapper.cs]: Create variable `assetPropertyTypeDto`

File: PortfolioKeeper/BusinessLogic/Mappers/AssetServiceMapper.cs [hunk 20]
```diff
@@ -33,12 +35,14 @@ public IEnumerable<AssetTypeDto> MapType(IEnumerable<AssetType> assetTypes)
                     PropertyTypes = assetPropertyTypeDtos
                 });
```
Comment (EdwOK) [inline PortfolioKeeper/BusinessLogic/Mappers/AssetServiceMapper.cs]: as well

**Pull request merged**